### ISSUE TITLE
RuntimeError: working outside of application context

### ISSUE THREAD

**Issue by rmasters** [marshmallow-code/flask-marshmallow#8]
I'm using Flask in the structure [defined here](https://www.digitalocean.com/community/tutorials/how-to-structure-large-flask-applications) and am coming across this error:

```
Traceback (most recent call last):
  File "run.py", line 2, in <module>
    from app import app
  File ".../app/__init__.py", line 13, in <module>
    from monitors.controllers import monitors
  File ".../app/monitors/controllers.py", line 3, in <module>
    from .models import Monitor, monitor_schema
  File ".../app/monitors/models.py", line 16, in <module>
    monitor_schema = MonitorSchema()
  File ".../site-packages/marshmallow/schema.py", line 227, in __init__
    self.opts = self.OPTIONS_CLASS(self.Meta)
  File ".../site-packages/flask_marshmallow/__init__.py", line 56, in __init__
    meta, 'strict', current_app.config['MARSHMALLOW_STRICT']
  File ".../site-packages/werkzeug/local.py", line 338, in __getattr__
    return getattr(self._get_current_object(), name)
  File ".../site-packages/werkzeug/local.py", line 297, in _get_current_object
    return self.__local()
  File ".../site-packages/flask/globals.py", line 34, in _find_app
    raise RuntimeError('working outside of application context')
RuntimeError: working outside of application context
```

I'm passing through the app instance as below (which should retain the app context?):
##### __init__.py:

``` python
app = Flask(__name__)
ma = Marshmallow(app)
```
##### monitors/models.py

``` python
from app import app, ma

class MonitorSchema(ma.Schema):
    class Meta:
        fields = ('id', 'name')

monitor_schema = MonitorSchema()
```
##### monitors/controllers.py

``` python
from .models import Monitor, monitor_schema

...
monitor = Monitor.query.get(1)
return monitor_schema.dump(monitor).data
```

Any ideas as to where I'm going wrong? Thanks

**Comment by rmasters** [comment]
It seems a way to work around it is to init the schema object under the app_context, but I'd like to know if I'm doing something ill-advised :)

##### monitors/models.py

``` python
from app import app, db, ma

class Model(db.Model):
    pass

class ModelSchema(ma.Schema):
    pass

with app.app_context():
    model_schema = ModelSchema()
    models_schema = ModelSchema(many=True)
```

**Comment by sloria** [comment]
@rmasters Indeed, the app context is necessary in order for flask-marshmallow to pull the app config values (`MARSHMALLOW_STRICT` and `MARSHALLOW_DATEFORMAT`).

We may deprecate these config options in a future release. Now that you've brought it up, I'm not sure that they add much value, since you can just create base schema class with the desired configuration.

**Comment by mr337** [comment]
+1 for this as this just bit me too

**Comment by melvinmt** [comment]
Thanks for fixing this, it works in the dev branch. Please consider upgrading the default installation when installing it through `pip install flask-marshmallow` as well.

**Comment by sloria** [comment]
This is now released to the PyPI (version 0.5.0).
